pennylane_snowflurry/monarq_device.py

```diff
@@ -0,0 +1,142 @@
+"""
+Contains the Device implementation of monarq.default
+"""
+
+from typing import Tuple
+from pennylane.devices import Device
+from pennylane.transforms import transform
+from pennylane.transforms.core import TransformProgram
+from pennylane.tape import QuantumScript, QuantumTape
+from pennylane_snowflurry.execution_config import DefaultExecutionConfig, ExecutionConfig
+from pennylane_snowflurry.API.adapter import ApiAdapter
+from pennylane_snowflurry.processing import PreProcessor, PostProcessor
+from pennylane_snowflurry.processing.config import ProcessingConfig, MonarqDefaultConfig
+from pennylane_snowflurry.API.client import ApiClient
+from pennylane_snowflurry.API.job import Job
+import pennylane.measurements as measurements
+
+
+class DeviceException(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+        
+
+class MonarqDevice(Device):
+    """PennyLane device for interfacing with Anyon's quantum Hardware.
+
+    * Extends the PennyLane :class:`~.pennylane.Device` class.
+    * Batching is not supported yet.
+
+    Args:
+        wires (int, Iterable[Number, str]): Number of wires present on the device, or iterable that
+            contains unique labels for the wires as numbers (i.e., ``[-1, 0, 2]``) or strings
+            (``['ancilla', 'q1', 'q2']``). Default ``None`` if not specified.
+        shots (int, Sequence[int], Sequence[Union[int, Sequence[int]]]): The default number of shots
+            to use in executions involving this device.
+        client (Client) : client information for connecting to MonarQ
+        behaviour_config (Config) : behaviour changes to apply to the transpiler
+    """
+
+    name = "CalculQCDevice"
+    short_name = "monarq.default"
+    pennylane_requires = ">=0.30.0"
+    author = "CalculQuebec"
+    
+    realm = "calculqc"
+    circuit_name = "test circuit"
+    project_id = ""
+    machine_name = "yamaska"
+
+    observables = {
+        "PauliZ"
+    }
+    
+    _processing_config : ProcessingConfig
+    
+    def __init__(self, 
+                 wires = None, 
+                 shots = None,  
+                 client : ApiClient = None,
+                 processing_config : ProcessingConfig = None) -> None:
+
+        if isinstance(shots, int) and (shots < 1 or shots > 1000) or isinstance(shots, list) and (len(shots) < 1 or len(shots) > 1000) or shots == None:
+            raise DeviceException("The number of shots must be contained between 1 and 1000")
```

Comment:
add context for when number of shots is None or the wrong type

Reply:
+ getting started doc doesnt reflect the fact that you need to pass a shot quantity